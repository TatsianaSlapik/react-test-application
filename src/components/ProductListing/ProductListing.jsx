import React, { useEffect } from "react";

import Product from "../Product/Product";
import Switcher from "../Switcher/Switcher";
import { connect, useDispatch } from "react-redux";
import actions from "../../utils/actions";

import "./ProductListing.scss";

function ProductListing({ listState, productsList }) {
  const dispatch = useDispatch();

  const setProductsList = (data) => {
    dispatch({ type: actions.PRODUCT_LIST, payload: data });
  };

  useEffect(() => {
    if (productsList.length === 0) {
      fetch(
        `https://fakerapi.it/api/v1/products?_quantity=10&_taxes=10&_categories_type=string`
      )
        .then((res) => res.json())
        .then((result) => {
          setProductsList(result.data);
        });
    }
  }, []);

  return (
    <div>
      <div className="product_switcher">
        <Switcher></Switcher>{" "}
      </div>
      <div
        className={listState ? "product_container list" : "product_container"}
      >
        {productsList.map((el, i) => {
          return (
            <Product
              name={el.name}
              description={el.description}
              image={el.image + `?id=${i}`}
              price={el.price}
              key={i}
              fullView={listState}
            ></Product>
          );
        })}
      </div>
    </div>
  );
}
const mapStateToProps = (state) => ({
  listState: state.listViewState,
  productsList: state.productsList,
});
export default connect(mapStateToProps)(ProductListing);
