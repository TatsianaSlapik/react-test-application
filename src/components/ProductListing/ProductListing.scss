.product_container {
  display: flex;
  flex-wrap: wrap;
  padding-top: 180px;
  justify-content: flex-end;
}
.list {
  flex-direction: column;
  padding-top: 90px;
  margin-left: 60px;
  margin-right: 60px;
}
