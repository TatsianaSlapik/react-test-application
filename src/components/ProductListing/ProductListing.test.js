import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import ProductListing from "./ProductListing.jsx";
const mockStore = configureStore([]);

describe("ProductListing", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      productsList: [],
      listViewState: false,
    });
    component = renderer.create(
      <Provider store={store}>
        <ProductListing />
      </Provider>
    );
  });

  it("Product Listing when listViewState is true should be rendered", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
