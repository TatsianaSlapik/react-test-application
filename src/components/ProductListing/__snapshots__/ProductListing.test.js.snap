// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListing Product Listing when listViewState is true should be rendered 1`] = `
<div>
  <div
    className="product_switcher"
  >
    <div
      className="toggle"
    >
      <input
        checked={false}
        className="toggle_input"
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="toggle_bg"
      />
      <div
        className="toggle_switch"
      >
        <div
          className="toggle_switch_grid"
        />
        <div
          className="toggle_switch_list"
        />
      </div>
    </div>
     
  </div>
  <div
    className="product_container"
  />
</div>
`;
